src/librustc/hir/lowering.rs

```diff
@@ -4738,16 +4738,10 @@ impl<'a> LoweringContext<'a> {
                     hir::MatchSource::ForLoopDesugar,
                 ));
 
-                // `{ let _result = ...; _result }`
-                // Underscore prevents an `unused_variables` lint if the head diverges.
-                let result_ident = self.str_to_ident("_result");
-                let (let_stmt, let_stmt_binding) =
-                    self.stmt_let(e.span, false, result_ident, match_expr);
-
-                let result = P(self.expr_ident(e.span, result_ident, let_stmt_binding));
-                let block = P(self.block_all(e.span, hir_vec![let_stmt], Some(result)));
-                // Add the attributes to the outer returned expr node.
-                return self.expr_block(block, e.attrs.clone());
+                // This is effectively `{ let _result = ...; _result }`.
+                // The construct was introduced in #21984.
```

Comment:
@pnkfelix Can you comment on whether this construct is actually relevant today? I haven't been able to encode a program where the difference can be observed with and without the construct when using the desugaring of `for`-loops. If it isn't relevant, then let's perhaps make an issue and remove it in a follow up PR.